pkg/ingester/ingester_v2.go

```diff
@@ -827,11 +799,7 @@ func (i *Ingester) v2Push(ctx context.Context, req *cortexpb.WriteRequest) (*cor
 
 		if i.cfg.ActiveSeriesMetricsEnabled && succeededSamplesCount > oldSucceededSamplesCount {
 			db.activeSeries.UpdateSeries(cortexpb.FromLabelAdaptersToLabels(ts.Labels), startAppend, func(l labels.Labels) labels.Labels {
-				// If we have already made a copy during this push, no need to create new one.
-				if copiedLabels != nil {
-					return copiedLabels
-				}
-				return cortexpb.CopyLabels(l)
+				return copiedLabels
```

Comment:
As an edge case, could happen `ts.Samples` is empty? If yes, then we would need to keep the previous logic because `copiedLabels` may be `nil`.

Reply:
If `ts.Samples` is empty then `succeededSamplesCount > oldSucceededSamplesCount` will be false and we will not arrive at this line.
I have added a comment to try to clarify.